src/components/SideBar.tsx

```diff
@@ -31,31 +31,73 @@ const SidebarLink = styled.a`
   }
 `
 
-interface Props {
-  activeSection: string | null
-  handleLinkClick: (sectionId: string) => void
+interface IobserverOptions {
+  root: HTMLElement | null
+  rootMargin: string
+  threshold: number | number[]
+}
+
+const defaultObserverOptions: IobserverOptions = {
+  root: null,
+  // when a section passes the area that is 10% from the top and 80% from the bottom of the viewport, it will be considered intersecting
+  rootMargin: '-20% 0px -80% 0px',
+  threshold: 0.0,
+}
+
+interface ISidebar {
+  handleLinkClick?: () => void
   sidebarList: string[]
+  sectionRefs: React.RefObject<Record<string, HTMLElement | null>>
+  observerOptions?: IobserverOptions
 }
 
 const capitalizeWords = (s: string) => {
```

Comment:
[4e49ce7](https://github.com/kixlab/website/pull/40/commits/4e49ce794d28db34102d9644cdf78275e3c033da)
One small issue I noticed is that we now have 'M S Student' and 'Ph D student' instead of 'M S. Student' and 'Ph D. Student'.